src/hooks/useL1CrossDomainMessengerProxy.ts

```diff
@@ -130,43 +109,42 @@ export function useL1CrossDomainMessengerProxy({
   const contract = getContract('L1CrossDomainMessengerProxy', fromChain.id)
   const { writeContractAsync } = useWriteContract()
 
-  const { gas: l2Gas, message } = useL2ContractCallInfo({
+  return useCallback(async () => {
+    const { gas: l2Gas, message } = await l2ContractCallInfo({
+      contractName,
+      functionName,
+      args,
+      value,
+      walletAddress,
+      chain: fromChain == sepolia ? optimismSepolia : optimism,
```

Comment:
how's that? Why `sepolia` means `optimismSepolia`?

I went to check the chains config (to see the contracts code) and got a bit confused with how's everything setup.

Why do we have:

```ts
  {
    abi: L1CrossDomainMessengerProxyABI,
    address: {
      [mainnet.id]: '0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1',
      [sepolia.id]: '0x58Cc85b8D04EA49cC6DBd3CbFFd00B4B8D6cb3ef',
    },
    name: 'L1CrossDomainMessengerProxy',
  },
```

and we use `optimismSepolia`/`optimism`?